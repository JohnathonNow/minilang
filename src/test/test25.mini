fun test1(N) do
	for I in 1 .. 10 do
		must print('I = {I}\n')
		for J in 1 .. 10 do
			must print('J = {I}\n')
			if (I * I) + (J * J) = (N * N) then
				exit exit '{I}^2 + {J}^2 = {N}^2'
			end
		end
	end
end

test1(5)

fun test2(F) do
	print('Starting\n')
	var I := 1
	must print('Must: I = {I}\n')
	I := 2
	I := F()
	print('Finishing: I = {I}\n')
	I := 3
end

do
	test2(;) 5
	test2(;) 1 / 0
on Error do
	print('{Error:type}: {Error:message}\n')
end
