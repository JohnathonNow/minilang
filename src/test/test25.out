J = 1
J = 1
J = 1
J = 1
J = 1
J = 1
J = 1
J = 1
J = 1
J = 1
I = 1
J = 2
J = 2
J = 2
J = 2
J = 2
J = 2
J = 2
J = 2
J = 2
J = 2
I = 2
J = 3
J = 3
J = 3
J = 3
I = 3
Starting
Finishing: I = 5
Must: I = 3
Starting
Must: I = 2
ValueError: Division by 0
